pkg/webhook/v1/notify/notify.go

```diff
@@ -0,0 +1,67 @@
+package notify
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/kubesphere/notification-manager/pkg/notify/config"
+	"github.com/prometheus/alertmanager/template"
+	"reflect"
+)
+
+type Notifier interface {
+	Notify(alert template.Alert) []error
+}
+
+type Factory func(logger log.Logger, config interface{}) Notifier
+
+var (
+	factorys map[string]Factory
+)
+
+func Register(name string, factory Factory) {
+	if factorys == nil {
+		factorys = make(map[string]Factory)
+	}
+
+	factorys[name] = factory
+}
+
+type Integration struct {
+	Notifiers []Notifier
+	Alert     template.Alert
+}
+
+func NewIntegration(logger log.Logger, receivers []*config.Receiver, alert template.Alert) *Integration {
```

Comment:
Why using word integration? It's strange.
Maybe Notification is better? Notification including notifiers and alerts sounds reasonable.